### PR TITLE
Add 'title' field to OperatingSystem entity

### PR DESCRIPTION
**Pull request by abalakh**
```python
>>> os1 = entities.OperatingSystem(name=gen_string('alphanumeric')).create()
2015-12-07 18:14:51 - nailgun.client - DEBUG - Making HTTP POST request to https://**************************.redhat.com:443/api/v2/operatingsystems with options {'verify': False, 'auth': ('*****', '********'), 'headers': {'content-type': 'application/json'}} and data {"operatingsystem": {"major": "5", "name": "co3zQU6cUT"}}.
2015-12-07 18:14:52 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":249,"name":"co3zQU6cUT","title":"co3zQU6cUT 5","description":null,"major":"5","minor":"","family":null,"release_name":null,"password_hash":"MD5","created_at":"2015-12-07T16:14:52Z","updated_at":"2015-12-07T16:14:52Z","parameters":[],"media":[],"architectures":[],"ptables":[],"config_templates":[],"os_default_templates":[],"images":[]}
>>> os1.title
u'co3zQU6cUT 5'
```

### CODE REVIEW

**Comment by Ichimonji10** [comment]
ACK

**Comment by omaciel** [comment]
APT

**Comment by oshtaier** [comment]
lol, just implemented similar thing in application itself - https://github.com/theforeman/foreman/pull/2925. In WebUI we call it 'description', but here in API we have all three of them - name, title and description.
Maybe, it is even a defect

**Comment by oshtaier** [comment]
btw, where did you find that parameter?
`apidoc/v2/operatingsystems/update.html` or `apidoc/v2/operatingsystems/create.html` doesn't have it

**Comment by oshtaier** [comment]
Please, investigate that. Till that moment NACK

**Comment by omaciel** [comment]
@oshtaier seems that this has been around since [Sept. 23, 2014](https://github.com/theforeman/foreman/commit/8b737c9c7648b3726dadb3b2e4708fcb43af02a8#diff-ca0b1ebc260d7633163da1e5450eabb9R8)

**Comment by abalakh** [comment]
@oshtaier @omaciel submitted [BZ1290359](https://bugzilla.redhat.com/show_bug.cgi?id=1290359) for that issue.
Seems 'title' is used more often than 'description', also 'description' is a bit buggy (unlike title it's empty by default, and if you set it to some value i think it just changes 'title' value). IMO 'description' should be dropped in favour of 'title', so adding 'title' field to nailgun isn't a bad thing.
@oshtaier , your thoughts?

**Comment by oshtaier** [comment]
@abalakh Sounds good. Please, provide corresponding comments in the code referring to that Bugzilla

**Comment by abalakh** [comment]
@oshtaier done, thanks for suggestion

File: nailgun/entities.py [hunk 1]
```diff
@@ -2457,6 +2457,8 @@ class OperatingSystem(
     * architecture
```
Comment (oshtaier) [inline nailgun/entities.py]: 1) `architecture`, `medium` and `ptable` are valid fields and are listed in the API doc for all three main operations - `GET`, `POST` and `PUT`
2) not sure that we need to mention that long URL `<https://bugzilla.redhat.com/show_bug.cgi?id=1290359>` as it was not mention 5 lines before
3) we should also mention `description` in some way

Comment (abalakh) [inline nailgun/entities.py]: 1) ok, will take a look and update
2) That bug 5 lines above is more like exclusion - take a brief look at whole file, in most cases we include full link. And also in the end it becomes beautiful short link like this:
2015-12-10 13_41_42-fedora 23 xfce light - vmware workstation
3) any ideas how?

Comment (oshtaier) [inline nailgun/entities.py]: nice

**Comment by oshtaier** [comment]
ACK non-blocking comments

**Comment by oshtaier** [comment]
ACK

**Pull request merged**